scdl/scdl.py

```diff
@@ -458,63 +456,116 @@ def download_track(track, playlist_name=None, playlist_file=None):
-            logging.info('{0} not large enough, skipping'.format(title))
-            return
+    if min_size is not None and total_length < min_size:
+        logging.info('{0} not large enough, skipping'.format(title))
+        return
 
-        if max_size is not None and total_length > max_size:
-            logging.info('{0} too large, skipping'.format(title))
-            return
+    if max_size is not None and total_length > max_size:
+        logging.info('{0} too large, skipping'.format(title))
+        return
 
-        with temp as f:
-            for chunk in progress.bar(
-                r.iter_content(chunk_size=1024),
-                expected_size=(total_length/1024) + 1,
-                hide=True if arguments["--hide-progress"] else False
-            ):
-                if chunk:
-                    f.write(chunk)
-                    f.flush()
-
-        shutil.move(temp.name, os.path.join(os.getcwd(), filename))
-        if filename.endswith('.mp3') or filename.endswith('.m4a'):
-            try:
-                set_metadata(track, filename, playlist_name)
-            except Exception as e:
-                logger.error('Error trying to set the tags...')
-                logger.debug(e)
-        else:
-            logger.error("This type of audio doesn't support tagging...")
+    with temp as f:
+        for chunk in progress.bar(
+            r.iter_content(chunk_size=1024),
+            expected_size=(total_length/1024) + 1,
+            hide=True if arguments["--hide-progress"] else False
+        ):
+            if chunk:
+                f.write(chunk)
+                f.flush()
+
+    shutil.move(temp.name, os.path.join(os.getcwd(), filename))
+    if filename.endswith('.mp3') or filename.endswith('.m4a'):
+        try:
+            set_metadata(track, filename, playlist_name)
+        except Exception as e:
+            logger.error('Error trying to set the tags...')
+            logger.debug(e)
+    else:
+        logger.error("This type of audio doesn't support tagging...")
 
-        #Try to change the real creation date
-        created_at = track['created_at']
-        filetime = int(time.mktime(datetime.strptime(created_at, '%Y/%m/%d %H:%M:%S %z').timetuple()))
-        try_utime(filename,filetime)
+    #Try to change the real creation date
+    created_at = track['created_at']
+    filetime = int(time.mktime(datetime.strptime(created_at, '%Y/%m/%d %H:%M:%S %z').timetuple()))
+    try_utime(filename,filetime)
 
-    else:
+    logger.info('{0} Downloaded.\n'.format(filename))
+    record_download_archive(track)
+
+
+def allow_download(track, title):
+    """
+    Returns True if the file should be downloaded
+    """
+    global arguments
+
+    # Not streamable
+    if not track['streamable']:
+        logger.error('{0} is not streamable...'.format(title))
+        return True
+
+    # Already downloaded
+    archive_fn = arguments.get('--download-archive')
```

Comment:
We should replace `archive_fn` by `archive_filename` which is more explicit.

Reply:
Done! e5869ce5418082d76a7b1eefb9e9a9112598e9d8